### PR TITLE
Make `from_waker`, `waker` and `from_raw` unstably `const`

### PR DESCRIPTION
**Pull request by BennoLossin**
Make
- `Context::from_waker`
- `Context::waker`
- `Waker::from_raw`

`const`.

Also added a small test.

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
r? @lcnr for wg-const-eval approval of new const-unstable APIs

(Closest reading of policy I could find suggests that wg-const-eval should approve, though I think these particular ones seem OK to my relatively uninformed opinion).

**Comment by lcnr** [comment]
these look fine for me, please add a tracking issue, then r=me

cc @rust-lang/wg-const-eval

**Review by oli-obk — APPROVED** [review]

**Review by fee1-dead — APPROVED** [review]

File: library/core/src/task/wake.rs [hunk 1]
```diff
@@ -186,17 +186,19 @@ pub struct Context<'a> {
 impl<'a> Context<'a> {
     /// Create a new `Context` from a [`&Waker`](Waker).
     #[stable(feature = "futures_api", since = "1.36.0")]
+    #[rustc_const_unstable(feature = "const_waker", issue = "none")]
```
Comment (lcnr) [inline library/core/src/task/wake.rs:189]: please also update the tracking issue here

**Pull request merged**